### DISCUSSION TITLE
Failed attempt to use eFinder during observing session on Dec 12, 2022

### DISCUSSION THREAD

**Discussion by WimDeMeester** [WimDeMeester/eFinder#35] [Show and tell (success stories and failures)]
Yesterday, we had a clear (but very cold!) night in Belgium, so I tested the eFinder for the second time.  Compared to the first test (on November 13), I changed the orientation of the camera and I updated the software to the latest version (the reworked version).
The test was not successful...  These were the problems:

- I tried to align the Nexus DSC Pro using the eFinder.  After aligning on two stars, I got the error message that the alignment could not be done because the two alignments were too close to each other...  Even when I moved the telescope a lot on the sky.  To fix this, I did a manual alignment using the Nexus.
- The alignment was quite good, and I pointed the telescope to the pleiades.  The pleiades were in the field of view of the telescope (it is difficult not to get them in the FoV as they have such a large diameter).  When I took an image using the eFinder, I could see the pleiades and I could solve the image correctly.  When I selected 'GoTo: via Align', the telescope was moved many degrees south of where it should move and the pleiades were moved very far out of the field of view...

![20221212-213944_0348+2414](https://user-images.githubusercontent.com/7261529/207288492-aa008332-8c50-4b4b-b1f8-e10864a1debc.jpg)

@AstroKeith Do you have any ideas what could have gone wrong here?

As a next test (hopefully later this week), I will try again with the previous version of the software.

**Comment by AstroKeith** [comment]
Oh dear!Firstly, there is a new bug in the Nexus. Serge is working on it. It affects two star alignments using the LX200 interface, ie SkySafari or eFinder.That said.  I Get that error when I have forgotten to take it out of test mode!I’m using 16_3 and getting good goto via align. Keith Sent from my iPhoneOn 13 Dec 2022, at 10:06, Wim De Meester ***@***.***> wrote:﻿
Yesterday, we had a clear (but very cold!) night in Belgium, so I tested the eFinder for the second time.  Compared to the first test (on November 13), I changed the orientation of the camera and I updated the software to the latest version (the reworked version).
The test was not successful...  These were the problems:

I tried to align the Nexus DSC Pro using the eFinder.  After aligning on two stars, I got the error message that the alignment could not be done because the two alignments were too close to each other...  Even when I moved the telescope a lot on the sky.  To fix this, I did a manual alignment using the Nexus.
The alignment was quite good, and I pointed the telescope to the pleiades.  The pleiades were in the field of view of the telescope (it is difficult not to get them in the FoV as they have such a large diameter).  When I took an image using the eFinder, I could see the pleiades and I could solve the image correctly.  When I selected 'GoTo: via Align', the telescope was moved many degrees south of where it should move and the pleiades were moved very far out of the field of view...


@AstroKeith Do you have any ideas what could have gone wrong here?
As a next test (hopefully later this week), I will try again with the previous version of the software.

**Comment by AstroKeith** [comment]
BTW the Nexus bug shouldn’t stop alignment. Just gives a bad one. The stars too close is another issue. Sent from my iPhoneOn 13 Dec 2022, at 10:23, Keith Venables ***@***.***> wrote:﻿Oh dear!Firstly, there is a new bug in the Nexus. Serge is working on it. It affects two star alignments using the LX200 interface, ie SkySafari or eFinder.That said.  I Get that error when I have forgotten to take it out of test mode!I’m using 16_3 and getting good goto via align. Keith Sent from my iPhoneOn 13 Dec 2022, at 10:06, Wim De Meester ***@***.***> wrote:﻿
Yesterday, we had a clear (but very cold!) night in Belgium, so I tested the eFinder for the second time.  Compared to the first test (on November 13), I changed the orientation of the camera and I updated the software to the latest version (the reworked version).
The test was not successful...  These were the problems:

I tried to align the Nexus DSC Pro using the eFinder.  After aligning on two stars, I got the error message that the alignment could not be done because the two alignments were too close to each other...  Even when I moved the telescope a lot on the sky.  To fix this, I did a manual alignment using the Nexus.
The alignment was quite good, and I pointed the telescope to the pleiades.  The pleiades were in the field of view of the telescope (it is difficult not to get them in the FoV as they have such a large diameter).  When I took an image using the eFinder, I could see the pleiades and I could solve the image correctly.  When I selected 'GoTo: via Align', the telescope was moved many degrees south of where it should move and the pleiades were moved very far out of the field of view...


@AstroKeith Do you have any ideas what could have gone wrong here?
As a next test (hopefully later this week), I will try again with the previous version of the software.

**Comment by WimDeMeester** [comment]
I'm now preparing a new SD-card for the Raspberry Pi with the newest version of the software, so I can switch easily between the old and the new versions...

**Comment by AstroKeith** [comment]
The 16_3 I’m using is much earlier than the repository version.Do you monitor the terminal window when you run the GUI? It can give a lot of clues. Sent from my iPhoneOn 13 Dec 2022, at 11:12, Wim De Meester ***@***.***> wrote:﻿
I'm now preparing a new SD-card for the Raspberry Pi with the newest version of the software, so I can switch easily between the old and the new versions...

**Reply by WimDeMeester** [reply]
> The 16_3 I’m using is much earlier than the repository version

Is the 16_3 you are using older than the one from https://github.com/WimDeMeester/eFinder/releases/tag/v16.3 ?

**Comment by AstroKeith** [comment]
Sorry - I wasnt clear!

The current repository version is way ahead.
Yes, the link you gave is to what I am using (or very close).

Keith

**Comment by WimDeMeester** [comment]
Found some problems in the code.  The solved_radec and solved_altaz variables were not global and as such resulted in 0, 0.  This explains why the alignment did not work (as the two stars were pointing to the same 0, 0 coordinates) and it also explains why the 'GoTo' problem.  
It might be clear tomorrow, so I hope to be able to test again.

**Comment by AstroKeith** [comment]
Excellent.

Serge hasn’t fixed the bug yet, so for now use the Nexus to do the two star alignment.

**Reply by WimDeMeester** [reply]
> Excellent. Serge hasn’t fixed the bug yet, so for now use the Nexus to do the two star alignment.

I'll do so.  But I guess I can first try the two stars alignment to see if the error about the two stars that are too close to each other is solved?

**Comment by AstroKeith** [comment]
That’s true!

But then repeat using the Nexus. The LX200 protocol alignment produces such a bad solution that the drives won’t work correctly.

**Comment by mrosseel** [comment]
a lot of the upcoming bugs will be caused by global variables, there are a
lot of them and not always clear where they're set - so it's easy to miss
one. Reducing them will make the code cleaner but introduce some bugs I
guess.

Mike

**Reply by WimDeMeester** [reply]
I'll take that in mind when I encounter new problems.  Is it correct that the  eFinder.py script was not adapted?  So, this script should work as before - I guess...

**Comment by mrosseel** [comment]
For now, yes. In the merge branch it's changed

**Reply by WimDeMeester** [reply]
But the merge branch is local on your account?

**Comment by AstroKeith** [comment]
Hi all,

Serge got back to me.

Nexus DSC - the bug is a stack overflow problem. New firmware required. He’s hope to get me something in a day and I will put it on test. The Nexus DSC is totally out of memory and so he has to borrow space from another stack. Hoping he doesnt just move the problem. Hence testing will be required before re-issue!

Nexus DSC Pro - the bug is currently unknown, but after 2 hours FaceTime today, Serge is seeing it on his set up. An easy workaround though - in Align set up, select 'two stars w/o Alt Ref’. Problem goes away.

Its all fun!

Keith

**Reply by WimDeMeester** [reply]
As I have the Nexus DSC Pro, I will use the 'two star w/o Alt Ref' option.  I guess I need to change that to use the 'Alt Ref' option again after Serge comes up with an updated firmware problem.

**Comment by mrosseel** [comment]
> But the merge branch is local on your account?

sure, that's not yet 'merged' to main :)

**Comment by AstroKeith** [comment]
Actually, the only benefit in using the Alt Ref, is the display of ‘angle’ below ’successful align’. If this is big, then it gives a good warning that an incorrect star was used, the encoders are slipping, or the scope mount moved.

W/o Alt Ref, you dont get this angle displayed.

**Comment by WimDeMeester** [comment]
After some new tests (see #41), I can confirm that the latest changes in the code fix the alignment and the GoTo problem I encountered during this night.
